diag_manager/fms_diag_dlinked_list.F90

```diff
@@ -41,282 +41,297 @@
+   !! a finalize keyword on the destructor, when this function returns and ll
+   !! goes out of scope, th allocations in initialized are undome
+   !! whether ot not ll is declared a pointer or allocatable
+   function linked_list_constructor () result (ll)
+      type(FmsDlList_t), pointer :: ll !< The resultant linked list to be reutrned.
+      allocate(ll)
+      call ll%initialize()
+   end function linked_list_constructor
 
-  !> @brief Constructor for the node_type
-  !! @return Returns a nully allocated node.
-  function node_constructor () result (nd)
-    type(FmsDlListNode_t), allocatable :: nd  !< The allocated node.
-    allocate(nd)
-    nd%data => null()
-    nd%prev => null()
-    nd%next => null()
-  end function node_constructor
+   !> @brief Initializer for the linked list.
+   !! @return Returns a newly allocated linked list instance.
+   subroutine linked_list_initializer( this )
+      class(FmsDlList_t), intent(inout) :: this  !<The instance of the class that this function is bound to
+      if( associated(this%head) .or. associated(this%tail)) then
+         call error_mesg('fms_diag_dlinked_list','linked list is already initalized', WARNING)
+      else
+         allocate(this%head)
+         allocate(this%tail)
+         this%head%next => this%tail
+         this%tail%prev => this%head
+         this%the_size = 0
+      endif
+   end subroutine linked_list_initializer
 
-  !> @brief Constructor for the linked list.
-  !! @return Returns a newly allocated linked list instance.
-  function linked_list_constructor () result (ll)
-    type(FmsDlList_t), allocatable :: ll !< The resultant linked list to be reutrned.
-    allocate(ll)
-    allocate(ll%head)
-    allocate(ll%tail)
-    !!print *, 'associated(ll%head) :' , associated(ll%head), &
-    !! ' associated(ll%head) :' , associated(ll%head)
-    ll%head%next => ll%tail
-    ll%tail%prev => ll%head
-    ll%the_size = 0
-  end function linked_list_constructor
 
-  !> @brief The list iterator constructor.
-  !! @return Returns a newly allocated list iterator.
-  function literator_constructor ( fnd, tnd ) result (litr)
-    type (FmsDlListNode_t), pointer  :: fnd
-    !< The sentinal (non-data) "first node" of the iterator will be fnd
-    type (FmsDlListNode_t), pointer  :: tnd
-    !< The sentinal (non-data) "last node" of the iterator will be tnd.
-    type (FmsDllIterator_t), allocatable :: litr  !< The resultant linked list to be reutrned.
-    allocate(litr)
-    litr%current => fnd
-    litr%end => tnd
-  end function literator_constructor
+   !> @brief The list iterator constructor.
+   !! @return Returns a newly allocated list iterator.
+   function literator_constructor ( fnd, tnd ) result (litr)
+      type (FmsDlListNode_t), pointer  :: fnd
+      !< The sentinal (non-data) "first node" of the iterator will be fnd
+      type (FmsDlListNode_t), pointer  :: tnd
+      !< The sentinal (non-data) "last node" of the iterator will be tnd.
```

Comment:
Will this style of commenting on the following line generate the documentation correctly?  I think these comments need to be on the same line as the variable, and you can use `!!` on the next line if there isnt enough space.

Reply:
It looks like this is in a lot of places.

Reply:
I think its working. See attached image made with 
``` ../configure --enable-docs && ``
![Screenshot from 2022-02-01 12-42-22](https://user-images.githubusercontent.com/42479054/152021818-94a37825-2e15-493a-89af-d7b1cc057fd7.png)
`